### PR TITLE
Various updates & fixes for a v2.0.0

### PR DESCRIPTION
**Pull request by UmanShahzad**
This commits various updates & fixes in preparation for a v2.0.0.

(It's a major version update because we make `buildHeaders` a private function).

### CODE REVIEW

**Comment by UmanShahzad** [comment]
We'll add support for timeouts and full Guzzle configurability in this PR as well.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 build
 composer.lock
 vendor
+psalm.xml
```
Comment (peter279k) [inline .gitignore:4]: I'm curious why the `psalm.xml`, the `Psalm` setting file is under `.gitignore` file.

Comment (UmanShahzad) [reply .gitignore]: Fixed in https://github.com/ipinfo/php/pull/21/commits/d276dcae072d326c6d23ef78ed40310378badf8e

**Comment by UmanShahzad** [comment]
Will release v2.0.0 to packagist as soon as I get access.

**Pull request merged**